.azure-pipelines/azure-pipelines.yml

```diff
@@ -0,0 +1,32 @@
+pool:
+  vmImage: 'ubuntu-latest'
+
+steps:
+- task: Maven@3
+  inputs:
+    mavenPomFile: 'pom.xml'
+    mavenOptions: '-Xmx3072m'
+    javaHomeOption: 'JDKVersion'
+    jdkVersionOption: '1.8'
+    jdkArchitectureOption: 'x64'
+    publishJUnitResults: true
+    testResultsFiles: '**/surefire-reports/TEST-*.xml'
+    goals: 'package'
```

Comment:
should this be `test`?

Reply:
I don't think it does any harm to test the entire packaging.

Reply:
Will this run the tests though? From the maven docs it looks like `package` just takes the compiled code and tries to bundle it into (in our case) a jar - http://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html

`mvn test` should handle both the compilation + testing step for us.

Reply:
Ahh I see that in the azure-pipelines run that tests did indeed get run, the output is a bit difficult to read, but in that case I don't have strong opinions about the `goal` definition